lib/node/runner/api/blocks.go

```diff
@@ -5,58 +5,67 @@ import (
 	"strconv"
 
 	"boscoin.io/sebak/lib/block"
-	"boscoin.io/sebak/lib/common"
 	"boscoin.io/sebak/lib/network/httputils"
 	"boscoin.io/sebak/lib/node/runner/api/resource"
 	"boscoin.io/sebak/lib/storage"
 )
 
 func (api NetworkHandlerAPI) GetBlocksHandler(w http.ResponseWriter, r *http.Request) {
-	p, err := NewPageQuery(r)
+	p, err := NewPageQuery(r, WithEncodePageCursor(false))
 	if err != nil {
 		httputils.WriteJSONError(w, err)
 		return
 	}
 
 	var (
-		firstCursor []byte
-		cursor      []byte // cursor as height
-		blocks      []resource.Resource
+		prevHeight uint64
+		nextHeight uint64
+		blocks     []resource.Resource
 	)
 
 	var option *storage.WalkOption
 	{
-		height, err := strconv.ParseUint(string(p.Cursor()), 10, 64)
-		if err != nil {
-			height = common.GenesisBlockHeight // default cursor is genesis block height
+		var cursor string
+		if p.Cursor() != nil {
+			height, err := strconv.ParseUint(string(p.Cursor()), 10, 64)
+			if err == nil {
```

Comment:
Shouldn't we return an error if the query contains an invalid cursor ?

Reply:
My intent is that it ignore wrong cursor param (error)

Reply:
Is that a good thing ? We usually error out when the query is wrong.

Reply:
Oh, it’s a good point. I will change it! Thanks @Geod24